### PR TITLE
[Build]: Fix the bin image generated from raw image issue

### PR DESCRIPTION
**Pull request by xumia**
#### Why I did it
It is to fix the issue https://github.com/Azure/sonic-buildimage/issues/10048
When building .raw image, for instance, target/sonic-broadcom.raw, it will generate a .bin image, target/sonic-broadcom.bin, as the intermediate file. The intermediate file is a build target which may contains different dependencies with the raw one.

#### How I did it
Rename the intermediate file. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [x] 202006
- [x] 202012
- [x] 202106
- [x] 202111

### CODE REVIEW

**Review by FuzailBrcm — APPROVED** [review]
Changes look good to me

**Comment by FuzailBrcm** [comment]
@Kalimuthu-Velappan 
Can you review these changes?

File: build_image.sh [hunk 1]
```diff
@@ -119,12 +121,13 @@ if [ "$IMAGE_TYPE" = "onie" ]; then
...
 
     echo "Build RAW image"
+    tmp_output_onie_image=${OUTPUT_ONIE_IMAGE}.tmp
     mkdir -p `dirname $OUTPUT_RAW_IMAGE`
     sudo rm -f $OUTPUT_RAW_IMAGE
 
     generate_device_list "./installer/$TARGET_PLATFORM/platforms_asic"
 
-    generate_onie_installer_image
+    generate_onie_installer_image "$tmp_output_onie_image"
```
Comment (liushilongbuaa) [inline build_image.sh:130]: It seems good also applying to kvm.

**Review by liushilongbuaa — APPROVED** [review]

**Comment by liuh-80** [comment]
Cherry pick PR created: https://github.com/sonic-net/sonic-buildimage/pull/13672

**Pull request merged**